packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -252,46 +400,172 @@ export class OwnedAddressTCRWithAppeals extends BaseWrapper<OwnedAddressTCRWithA
+      if (!revealPeriodActive) {
+        isInRevealPhase = false;
+      }
     }
 
     return isInRevealPhase;
   }
 
+  /**
+   * Checks if a listing is currently in the Request Appeal phase
+   * @param listingAddress Address of listing to check
+   */
+  public async isInRequestAppealPhase(listingAddress: EthAddress): Promise<boolean> {
+    const appealExpiryDate = await this.getRequestAppealExpiryDate(listingAddress);
+    const hasRequestedAppeal = await this.instance.getAppealRequested.callAsync(listingAddress);
+    return (!hasRequestedAppeal && appealExpiryDate > new Date());
+  }
+
+  /**
+   * Gets the expiry date-time of the request appeal phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getRequestAppealExpiryDate(listingAddress: EthAddress): Promise<Date> {
+    const requestAppealExpiryTimestamp = await this.getRequestAppealExpiryTimestamp(listingAddress);
+    return new Date(requestAppealExpiryTimestamp.toNumber() * 1000);
+  }
+
+  /**
+   * Checks if a listing is currently in the Appeal phase
+   * @param listingAddress Address of listing to check
+   */
+  public async isInAppealPhase(listingAddress: EthAddress): Promise<boolean> {
+    const appealExpiryDate = await this.getAppealExpiryDate(listingAddress);
+    return (appealExpiryDate > new Date());
+  }
+
+  /**
+   * Gets the expiry time of the appeal phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getAppealExpiryDate(listingAddress: EthAddress): Promise<Date> {
+    const appealExpiryTimestamp = await this.instance.getAppealPhaseExpiry.callAsync(listingAddress);
+    return new Date(appealExpiryTimestamp.toNumber() * 1000);
+  }
+
+  /**
+   * Gets the expiry time of the commit vote phase.
+   * @param listingAddress Address of listing to check
+   * @throws {CivilErrors.NoChallenge}
+   */
+  public async getCommitVoteExpiryDate(listingAddress: EthAddress): Promise<Date> {
+    // if there is no challenge
+    const challenge = await this.instance.getListingChallengeID.callAsync(listingAddress);
+    if (challenge.isZero()) {
+      throw CivilErrors.NoChallenge;
+    }
+    const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
+    const revealPeriodActive = await voting.getCommitPeriodExpiry(challenge);
+
+    return new Date(revealPeriodActive.toNumber() * 1000);
+  }
+
+  /**
+   * Gets the expiry time of the commit vote phase.
+   * @param listingAddress Address of listing to check
+   * @throws {CivilErrors.NoChallenge}
+   */
+  public async getRevealVoteExpiryDate(listingAddress: EthAddress): Promise<Date> {
+    // if there is no challenge
+    const challenge = await this.instance.getListingChallengeID.callAsync(listingAddress);
+    if (challenge.isZero()) {
+      throw CivilErrors.NoChallenge;
+    }
+    const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
```

Comment:
use `this.getVotingAddress()`.

actually, might as well create `this.getVoting(): Promise<Voting>`